pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -224,6 +224,16 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				}
 			}
 
+			con.mu.Lock()
```

Comment:
I know, you just moved the code from below, but what do you think about refactoring it as follows - I am just thinking taking a unconditional lock and releasing it based on conditions in if else might lead to subtle bugs if conditions change?

```
        bool added := false
	con.mu.Lock()
        added = con.added
        con.added = true
      con.mu.Unlock()
      if !added {
	s.addCon(con.ConID, con)
	defer s.removeCon(con.ConID, con)
      }
```